tests/bot/cogs/moderation/test_utils.py

```diff
@@ -0,0 +1,411 @@
+import textwrap
+import unittest
+from datetime import datetime
+from unittest.mock import AsyncMock, MagicMock, call, patch
+
+from discord import Embed, Forbidden, HTTPException, NotFound
+
+from bot.api import ResponseCodeError
+from bot.cogs.moderation import utils
+from bot.constants import Colours, Icons
+from tests.helpers import MockBot, MockContext, MockMember, MockUser
+
+
+class ModerationUtilsTests(unittest.IsolatedAsyncioTestCase):
+    """Tests Moderation utils."""
+
+    def setUp(self):
+        self.bot = MockBot()
+        self.member = MockMember(id=1234)
+        self.user = MockUser(id=1234)
+        self.ctx = MockContext(bot=self.bot, author=self.member)
+
+    async def test_user_get_active_infraction(self):
+        """
+        Should request the API for active infractions and return infraction if the user has one or `None` otherwise.
+
+        A message should be sent to the context indicating a user already has an infraction, if that's the case.
+        """
+        test_cases = [
+            {
+                "get_return_value": [],
+                "expected_output": None,
+                "infraction_nr": None,
+                "send_msg": True
+            },
+            {
+                "get_return_value": [{"id": 123987}],
+                "expected_output": {"id": 123987},
+                "infraction_nr": "123987",
+                "send_msg": False
+            },
+            {
+                "get_return_value": [{"id": 123987}],
+                "expected_output": {"id": 123987},
+                "infraction_nr": "123987",
+                "send_msg": True
+            }
+        ]
+
+        for case in test_cases:
+            with self.subTest(return_value=case["get_return_value"], expected=case["expected_output"]):
+                self.bot.api_client.get.reset_mock()
+                self.ctx.send.reset_mock()
+
+                params = {
+                    "active": "true",
+                    "type": "ban",
+                    "user__id": str(self.member.id)
+                }
+
+                self.bot.api_client.get.return_value = case["get_return_value"]
+
+                result = await utils.get_active_infraction(self.ctx, self.member, "ban", send_msg=case["send_msg"])
+                self.assertEqual(result, case["expected_output"])
+                self.bot.api_client.get.assert_awaited_once_with("bot/infractions", params=params)
+
+                if case["send_msg"] and case["get_return_value"]:
+                    self.ctx.send.assert_awaited_once()
+                    self.assertTrue(case["infraction_nr"] in self.ctx.send.call_args[0][0])
+                    self.assertTrue("ban" in self.ctx.send.call_args[0][0])
```

Comment:
Can use `assertIn` to get a much nicer error message if the test fails.

Perhaps assigning `self.ctx.send.call_args[0][0]` to a variable first would greatly improve readability?
```python
self.assertIn("ban", sent_message)
```

Reply:
Used here `assertIn` instead and in one other location.